src/main/scala/com/databricks/labs/overwatch/pipeline/Pipeline.scala

```diff
@@ -425,6 +427,33 @@ class Pipeline(
     Helpers.getDatesGlob(startDate.toLocalDate, untilTime.asLocalDateTime.plusDays(1).toLocalDate)
   }
 
+  //Get the dataframe from either tempDirectory or from the ResponseArray
+  private def getApiData(apiTempPath: Option[String]): DataFrame ={
+     //Get the data from either tempDirectory or from the ResponseArray
+     val tmpClusterEventsSuccessPath = s"${config.tempWorkingDir}/${apiTempPath.get}/success" + config.runID
+    val tmpClusterEventsErrorPath = s"${config.tempWorkingDir}/${apiTempPath.get}/error" + config.runID
+    null
+  }
+
+  //Add moduleID to the dataframe
+  private def addModuleID(apiDF: DataFrame):DataFrame = {
+    null
+  }
+
+  //Transform the api data
+  // convert responses to binary
+  // add moduleID to the DF
+  private def transformApiData(apiDF: DataFrame):DataFrame = {
+    addModuleID(apiDF)
+    //convert responses to binary
+  }
+  //Persist the Api Data
+  private def persistApiData(target: PipelineTable, module: Module): Unit = {
+      val apiDF  = getApiData(target.apiEndpoint)
+      val finalDF = transformApiData(apiDF)
```

Comment:
I really don't think we need two funcs here. It's a cast and a withCol -- suggest you just do it here.

Also -- these signatures won't actually allow you to get ta DF.

Reply:
done